pallets/asset/src/lib.rs

```diff
@@ -362,22 +380,28 @@ decl_module! {
 
             // Migrate `AssetDocuments`.
             use frame_support::Blake2_128Concat;
-            use polymesh_primitives::migrate::{migrate_double_map, Migrate};
-            use polymesh_primitives::document::DocumentOld;
+            use polymesh_primitives::{ migrate::{migrate_double_map, Migrate}, document::DocumentOld};
             use sp_std::collections::btree_map::BTreeMap;
-            let mut id_map = BTreeMap::<_, u32>::new();
-            migrate_double_map::<_, _, Blake2_128Concat, _, _, _, _, _>(
+
+
+            let storage_ver = <StorageVersion>::get();
+            if storage_ver == Version::V0 {
+                let mut id_map = BTreeMap::<_, u32>::new();
+                migrate_double_map::<_, _, Blake2_128Concat, _, _, _, _, _>(
                 b"Asset", b"AssetDocuments",
-                |ticker: Ticker, name: DocumentName, doc: DocumentOld| {
-                    let count = id_map.entry(ticker).or_default();
-                    let id = DocumentId(mem::replace(count, *count + 1));
-                    Some((ticker, id, doc.migrate(name)?))
+                    |ticker: Ticker, name: DocumentName, doc: DocumentOld| {
+                        let count = id_map.entry(ticker).or_default();
+                        let id = DocumentId(mem::replace(count, *count + 1));
+                        Some((ticker, id, doc.migrate(name)?))
+                    }
+                    );
+                for (ticker, id) in id_map {
+                    AssetDocumentsIdSequence::insert(ticker, DocumentId(id));
                 }
-            );
-            for (ticker, id) in id_map {
-                AssetDocumentsIdSequence::insert(ticker, DocumentId(id));
             }
 
+            <StorageVersion>::put(Version::V1);
```

Comment:
Could be inside the if loop otherwise it will re-write the version every time for those pallets which already at that given version.

Reply:
It is not going to be re-written because `on_runtime_upgrade` is not going to be executed.

Reply:
I think it will if we left the code within this pallet and try to upgrade the code for another pallet, Which triggers the `on_runtime_upgrade` of every pallet, and then it will lead to re write